### PR TITLE
Introduced more resilient exception handling.

### PR DESCRIPTION
**Pull request by HugaertsDries**
Small changes to make the exception handeling more robust.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -5,27 +5,29 @@ import { run as mailHealingReport } from './tasks/mail-healing-report';
...
-    try {
-      await mailHealingReport(inserts);
-      await mailOnError(inserts);
-      await mailOnFailedJob(inserts);
+  if (inserts.length) {
+    for (const run of [mailHealingReport, mailOnError, mailOnFailedJob]) {
+      try {
+        await run(inserts);
+      } catch (error) {
+        console.error(`There was an error processing ${inserts.join('\n')}`);
```
Comment (cecemel) [inline app.js]: can we log an error here too?

**Comment by cecemel** [comment]
@HugaertsDries https://github.com/lblod/delta-producer-report-generator/blob/master/app.js#L25 I see no response being send back, might lead to instability too. Can you have a look?

**Comment by HugaertsDries** [comment]
So I've improved some items after the initial commit:
- when an unexpected exception happens, it is logged as an error-alert
- the service releases the delta endpoint as fast as possible and always closes properly
- the service "silently" logs if the processing of one healing task fails & puts it on "failed" (if you want an error-alert, let me know)

**Comment by HugaertsDries** [comment]
To test I recommend the following:

1. Create a Task without a properly constructed Job
2. Make the delta call

Result: 204 BUT a warning in the logs that it could not create a report because the Job was null or couldn't be destructed

1. Create a Task with a properly constructed Job
2. Make the delta call

Result: 204 should work just fine, as before
 
To create the unexpected exception, tips are welcome but for now I actually break the code and see if it prepares and eventually sends out an email.

Some helpful queries:
```sql
PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
PREFIX task: <http://redpencil.data.gift/vocabularies/tasks/>
PREFIX dct: <http://purl.org/dc/terms/>
PREFIX prov: <http://www.w3.org/ns/prov#>
PREFIX nie: <http://www.semanticdesktop.org/ontologies/2007/01/19/nie#>
PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
PREFIX oslc: <http://open-services.net/ns/core#>
PREFIX cogs: <http://vocab.deri.ie/cogs#>
PREFIX adms: <http://www.w3.org/ns/adms#>
PREFIX nfo: <http://www.semanticdesktop.org/ontologies/2007/03/22/nfo#>
PREFIX dbpedia: <http://dbpedia.org/resource/>
PREFIX nmo: <http://www.semanticdesktop.org/ontologies/2007/03/22/nmo#>

# DELETE TASK
DELETE {
  GRAPH <http://mu.semte.ch/graphs/organizations/141d9d6b-54af-4d17-b313-8d1c30bc3f5b/LoketAdmin> {
     <http://mu.semte.ch/tasks/5daae0c1-dbc1-4f67-b9ec-9c6b2609478d> ?p ?o .
  }
} WHERE {
 GRAPH <http://mu.semte.ch/graphs/organizations/141d9d6b-54af-4d17-b313-8d1c30bc3f5b/LoketAdmin> {
    <http://mu.semte.ch/tasks/5daae0c1-dbc1-4f67-b9ec-9c6b2609478d> ?p ?o .
 }
}
;
# INSERT TASK
INSERT DATA {
    GRAPH <http://mu.semte.ch/graphs/organizations/141d9d6b-54af-4d17-b313-8d1c30bc3f5b/LoketAdmin> {
        <http://mu.semte.ch/tasks/5daae0c1-dbc1-4f67-b9ec-9c6b2609478d>
            a <http://redpencil.data.gift/vocabularies/tasks/Task> ;
            mu:uuid """5daae0c1-dbc1-4f67-b9ec-9c6b2609478d""" ;
            dct:isPartOf <http://mu.semte.ch/jobs/28bfdb49-0797-498a-a5e0-8e2788d5deea> ;
            dct:created "2021-10-26T22:32:52"^^xsd:dateTime ;
            dct:modified "2021-10-26T22:32:52"^^xsd:dateTime ;
            adms:status <http://redpencil.data.gift/id/concept/JobStatus/scheduled> ;
            task:inputContainer <http://mu.semte.ch/container/c1f27660-d04e-4922-bdc2-7ebe0747877b> ;
            task:index "1" ;
            task:operation <http://redpencil.data.gift/id/jobs/concept/TaskOperation/deltas/healing/reportGeneration> .

        <http://mu.semte.ch/container/c1f27660-d04e-4922-bdc2-7ebe0747877b>
            task:hasFile <http://mu.semte.ch/report/27a028e7-34ab-4e7e-bc4d-2eaad503d09e>.
    }
}
;
# DELETE TASK
DELETE {
  GRAPH <http://mu.semte.ch/graphs/organizations/141d9d6b-54af-4d17-b313-8d1c30bc3f5b/LoketAdmin> {
     <http://mu.semte.ch/jobs/28bfdb49-0797-498a-a5e0-8e2788d5deea>  ?p ?o .
  }
} WHERE {
 GRAPH <http://mu.semte.ch/graphs/organizations/141d9d6b-54af-4d17-b313-8d1c30bc3f5b/LoketAdmin> {
    <http://mu.semte.ch/jobs/28bfdb49-0797-498a-a5e0-8e2788d5deea>  ?p ?o .
 }
}
;
# INSERT JOB
INSERT DATA {
    GRAPH <http://mu.semte.ch/graphs/organizations/141d9d6b-54af-4d17-b313-8d1c30bc3f5b/LoketAdmin> {
        <http://mu.semte.ch/jobs/28bfdb49-0797-498a-a5e0-8e2788d5deea>
            a <http://vocab.deri.ie/cogs#Job> ;
            mu:uuid """28bfdb49-0797-498a-a5e0-8e2788d5deea""" ;
            dct:creator <http://mu.semte.ch/services/healing>;
            adms:status <http://redpencil.data.gift/id/concept/JobStatus/failed>;
            dct:created "2021-10-26T22:32:52"^^xsd:dateTime ;
            dct:modified "2021-10-26T22:32:52"^^xsd:dateTime ;
            task:operation <http://redpencil.data.gift/id/jobs/concept/TaskOperation/deltas/healing/reportGeneration> .
    }
}
```

Helpful body payload:
```json
    [
      { "inserts": [
            {
                "subject": { "type": "uri", "value": "http://mu.semte.ch/tasks/5daae0c1-dbc1-4f67-b9ec-9c6b2609478d" },
                "predicate": { "type": "uri", "value": "http://www.w3.org/ns/adms#status" },
                "object": { "type": "uri", "value": "http://redpencil.data.gift/id/concept/JobStatus/scheduled" }
            }
        ],
        "deletes": [] 
        }
    ]
```

**Comment by HugaertsDries** [comment]
Lastly, the build fails? Any clue as to why?

@cecemel @claire-lovisa

**Comment by cecemel** [comment]
@HugaertsDries thanks, for me fine, can be released and bumped. Let's try it in the loket stack first

**Pull request merged**